### ISSUE TITLE
certain require do not load extension Service class

### ISSUE THREAD

**Issue by mkristian** [jruby/jruby#2986]
```
$ bin/jruby -e "require 'uri:classloader:/lib/ruby/shared/json/ext/parser'; p JSON::Ext::Parser"
```
fails. same with the *file:*, *jar:*, *file:jar:* protocol. the *uri:jar:file:my.jar!/" protocol does not even look into the right place place to load the Service class from the gem extension.

**Comment by mkristian** [comment]
I was wrong about the **uri:file:* protocol and the wrong place. is part of the test now. closing.

**Comment by mkristian** [comment]
on master this is already fixed with 764d6af8

@headius either I revert this patch and backport master to jruby-1_7 or do the other way around. please also have a look at #2987 

in any case I would cherry-pick the tests added with 62fa282
